concrete/src/Multilingual/Service/Detector.php

```diff
@@ -33,7 +32,7 @@ public static function getPreferredSection()
         if ($result === null) {
             $locale = false;
             // Detect locale by value stored in session or cookie
-            if ($session->has('multilingual_default_locale')) {
+            if (self::canSetSessionValue() && $session->has('multilingual_default_locale')) {
```

Comment:
canSetSessionValue is not a static method (and I prefer it's not a static method), so you have to write `$this->canSetSessionValue()` instead of  `self::canSetSessionValue()`

Reply:
`canSetSessionValue()` is not but `getPreferredSection()` is and is called in a non object context.
I can at least avoid it in the `setupSiteInterfaceLocalization()` method.

Reply:
You need to get an instance of the class before calling its non static methods. See for example https://3v4l.org/dLKCA

Reply:
Ok I see what you mean, in this case is it better to use directly `$detector = new Detector()` or to use `$detector = $app->make(Detector::class)` ?

Reply:
I'd use the second

Reply:
Ok, thank you for your help !